source/extensions/http/original_ip_detection/xff/xff.cc

```diff
@@ -11,7 +11,9 @@ namespace Xff {
 
 XffIPDetection::XffIPDetection(
     const envoy::extensions::http::original_ip_detection::xff::v3::XffConfig& config)
-    : xff_num_trusted_hops_(config.xff_num_trusted_hops()), append_xff_(config.append_xff()) {
+    : xff_num_trusted_hops_(config.xff_num_trusted_hops()),
+      skip_xff_append_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(
```

Comment:
Sorry to introduce yet another new change.

In updating the proto docs for `append_xff` I realised it should probably be named consistently with the equivalent HCM option, `skip_xff_append`, so I've changed that, but this has a somewhat unfortunate consequence: Ideally Envoy should always append XFF by default, but the xff extension currently doesn't do that and there are dozens of tests that rely on this behaviour, and I imagine Envoy users also may be depending on this behaviour.

I've therefore made this default to true for users of the existing `xff_num_trusted_hops` setting, but for the new `xff_trusted_cidrs` option I would prefer to disable this setting by default to provide proper proxy behaviour out of the box.

Let me know if you'd prefer I back this out or make the setting consistent. The current behaviour is documented in the proto so it's prominent in the docs.

Reply:
ugh, yeah.  Generally I'm in favor of "fixing" broken or inconsistent config with a runtime guard for temporary workaround but I think in this case the risk is too high.  Checking with @mattklein123 for a second opinion but I'm inclined to agree on playing it safe over consistent config.